Add information on how to install wanda directly for development (#178)
trento-project/wanda: guides/development/hack_on_wanda.md

# Hack on Wanda

In order to run Wanda, the following is needed

1. [Elixir](https://elixir-lang.org/), a programming language used to build scalable and maintainable applications.
2. [Docker](https://docs.docker.com/get-docker/), a platform for developing, shipping, and running applications in containers.
3. [Docker Compose](https://docs.docker.com/compose/install/), a tool for defining and running multi-container applications.

## Development environment

A `docker-compose` development environment is provided for easy setup and management of dependencies. To start the development environment, simply run the following command:

```
docker-compose up -d
```

This command starts a **Postgres** database and a **RabbitMq** instance, which is used for storage and communication respectively.

## Setup Wanda

> After installing the dependencies, you need to perform some initial setup tasks before you can start Wanda. You can do this by running the following command:

Reviewer: Maybe this needs revisiting as the dependencies are not installed yet at this point.
```suggestion
Before starting Wanda, some initial setup tasks are required. This can be achieved by running the following command:
```

```
mix setup
```

> This command performs necessary tasks such as creating the database schema and running migrations.

Reviewer: We can add a reference to dependencies installation here maybe.

```suggestion
This command performs necessary tasks such as installing dependencies, creating the database schema and running migrations.
```

### Hint about Project setup

> Gain a deeper understanding of how Wanda is configured, reading the [mix.exs](/mix.exs) file located in the root directory of the project. This file contains information on dependencies, configuration settings, and tasks that can be run using the Mix build tool, providing a complete picture of the project's setup.

Reviewer: Let' use an absolute URL to `mix.exs`  as well.

> Gain a deeper understanding of how Wanda is configured, reading the mix.exs file located in the root directory of the project. This file contains information on dependencies, configuration settings, and tasks that can be run using the Mix build tool, providing a complete picture of the project's setup.

Reviewer 2: maybe add a link here to the actual `mix.exs` file?

# Hack on Wanda

> In order to run Wanda, you will need

Reviewer: ```suggestion
In order to run Wanda, the following is needed
```

1. [Elixir](https://elixir-lang.org/), a programming language used to build scalable and maintainable applications.
2. [Docker](https://docs.docker.com/get-docker/), a platform for developing, shipping, and running applications in containers.
3. [Docker Compose](https://docs.docker.com/compose/install/), a tool for defining and running multi-container applications.

## Development environment

A `docker-compose` development environment is provided for easy setup and management of dependencies. To start the development environment, simply run the following command:

```
docker-compose up -d
```

This command will start a **Postgres** database and a **RabbitMq** instance, which will be used for storage and communication respectively.

## Install dependencies

Before you can start hacking on Wanda, you will need to install its dependencies. You can do this by running the following command:

```
mix deps.get
```

This command will install all the required libraries and dependencies for the application.

## Setup Wanda

After installing the dependencies, you will need to perform some initial setup tasks before you can start Wanda. You can do this by running the following command:

```

> mix setup

Reviewer: Keep in mind that `mix setup` also runs `deps.get`. 

So we might want to either drop the previous `mix deps.get` section or replace `mix setup` with `ecto.setup`.

In any case I would also give a reference to `mix.exs` for further details, in order to avoid documenting every single task, which very likely we'll forget.

Author: Hey yeah this is absolutely correct. I removed a section and added a hint about the mix.exs
You are fully right!
